src/webui/api/torrentscontroller.cpp

```diff
@@ -622,6 +622,26 @@ void TorrentsController::setDownloadLimitAction()
     applyToTorrents(hashes, [limit](BitTorrent::TorrentHandle *torrent) { torrent->setDownloadLimit(limit); });
 }
 
+void TorrentsController::setShareLimitAction()
+{
+    checkParams({"hashes", "ratioLimit", "seedingTimeLimit"});
+
+    qreal ratioLimit = params()["ratioLimit"].toDouble();
+    if (ratioLimit == 0)
+        ratioLimit = -1;
+
+    qlonglong seedingTimeLimit = params()["seedingTimeLimit"].toLongLong();
+    if (seedingTimeLimit == 0)
```

Comment:
use `<=`?
And also in above instance.

Reply:
Why do you drop zero?

FYI, setRatioLimit/setSeedingTimeLimit checks its params for incorrect values.

Reply:
I was trying to handle the case where values of 0 should be treated as no limit. But the GUI doesn't seem to exhibit this behavior, so removing this code.

Reply:
Actually, it does do this in `UpDownRatioDlg::ratio()`. But this is WebUI-implementation specific so I'm now handling this in `shareratio.html`'s js logic.